Wrappers/Python/ccpi/optimisation/algorithms/GradientDescent.py

```diff
@@ -44,11 +44,12 @@ def __init__(self, x_init=None, objective_function=None, rate=None, **kwargs):
         :param x_init: initial guess
         :param objective_function: objective function to be minimised
         :param rate: step rate
+        :param alpha: optional parameter to start the backtracking algorithm
```

Comment:
We should add an additional parameter `beta`, which is typically used, and which governs how aggressively the step size is being reduced.  The current choice corresponds to `beta=0.5`, which is a fine default choice, but we should allow `beta` to be in the range from 0 to 1, both excluded. The only change, I think, would be to change the line in which `alpha` is updated to be `self.alpha = self.alpha*self.beta`.